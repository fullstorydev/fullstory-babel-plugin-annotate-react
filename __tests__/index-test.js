const babel = require('babel-core');
const plugin = require('../');
const assert = require('assert');

it('unknown-element snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component } from 'react';

class componentName extends Component {
  render() {
    return <bogus><h1>A</h1></bogus>;
  }
}

export default componentName;
`,
    {
      filename: "./filename-test.js",
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});

it('component-fragment snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component, Fragment } from 'react';

class componentName extends Component {
  render() {
    return <Fragment>A</Fragment>;
  }
}

export default componentName;
`,
    {
      filename: "./filename-test.js",
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});


it('component-react-fragment snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component } from 'react';

class componentName extends Component {
  render() {
    return <React.Fragment>A</React.Fragment>;
  }
}

export default componentName;
`,
    {
      filename: "./filename-test.js",
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});

it('arrow-noreturn-fragment snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component, Fragment } from 'react';

const componentName = () => (
  <Fragment>
    <h1>Hello world</h1>
  </Fragment>
);

export default componentName;
`,
    {
      filename: "./filename-test.js",
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});


it('rawfunction-fragment snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component, Fragment } from 'react';

function SubComponent() {
  return <Fragment>Sub</Fragment>;
}

const componentName = () => {
  return <Fragment>
    <SubCoponent />
  </Fragment>;
};

export default componentName;
`,
    {
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});


it('arrow snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component } from 'react';

const componentName = () => {
  return <div>
    <h1>Hello world</h1>
  </div>;
};

export default componentName;
`,
    {
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});


it('option-attribute snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component } from 'react';

const componentName = () => {
  return <div>
    <h1>Hello world</h1>
  </div>;
};

export default componentName;
`,
    {
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});


it('component snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component } from 'react';

class componentName extends Component {
  render() {
    return <div>
        <h1>Hello world</h1>
      </div>;
  }
}

export default componentName;
`,
    {
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});


it('rawfunction-react-fragment snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component } from 'react';

function SubComponent() {
  return <React.Fragment>Sub</React.Fragment>;
}

const componentName = () => {
  return <React.Fragment>
    <SubCoponent />
  </React.Fragment>;
};

export default componentName;
`,
    {
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});


it('arrow-anonymous-react-fragment snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component } from 'react';

const componentName = () => {
  return (() => <React.Fragment>
    <h1>Hello world</h1>
  </React.Fragment>)();
};

export default componentName;
`,
    {
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});


it('arrow-noreturn snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component } from 'react';

const componentName = () => (
  <div>
    <h1>Hello world</h1>
  </div>
);

export default componentName;
`,
    {
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});


it('arrow-noreturn-react-fragment snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component } from 'react';

const componentName = () => (
  <React.Fragment>
    <h1>Hello world</h1>
  </React.Fragment>
);

export default componentName;
`,
    {
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});


it('tags snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component } from 'react';
import { StyleSheet, Text, TextInput, View, Image, UIManager } from 'react-native';

UIManager.getViewManagerConfig('RCTView').NativeProps.fsClass = "String";

class Bananas extends Component {
  render() {
    let pic = {
      uri: 'https://upload.wikimedia.org/wikipedia/commons/d/de/Bananavarieties.jpg'
    };
    return <Image source={pic} style={{ width: 193, height: 110, marginTop: 10 }} fsClass="test-class" />;
  }
}

class PizzaTranslator extends Component {
  constructor(props) {
    super(props);
    this.state = { text: '' };
  }

  render() {
    return <View style={{ padding: 10 }}>
        <TextInput style={{
        backgroundColor: '#000',
        color: '#eee',
        padding: 8
      }} placeholder="Type here to translate!" // not supported on iOS
      onChangeText={text => this.setState({ text })} value={this.state.text} />
        <Text style={{ padding: 10, fontSize: 42 }}>
          {this.state.text.split(' ').map(word => word && 'üçï').join(' ')}
        </Text>
      </View>;
  }
}

export default function App() {
  return <View style={styles.container}>
      <Text style={{ color: '#eee' }}>FullStory ReactNative testing app</Text>
      <Bananas />
      <PizzaTranslator />
    </View>;
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'stretch',
    backgroundColor: '#222',
    alignItems: 'center',
    justifyContent: 'center'
  }
});
`,
    {
      filename: "./filename-test.js",
      presets: ["@babel/preset-react"],
      plugins: [
        [plugin, { native: true }]
      ]
    },
  );
  expect(code).toMatchSnapshot();
});


it('option-format snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component } from 'react';

const componentName = () => {
  return <div>
    <h1>Hello world</h1>
  </div>;
};

export default componentName;
`,
    {
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});


it('pureComponent-react-fragment snapshot matches', () => {
  const { code } = babel.transform(
`import React from 'react';

class PureComponentName extends React.PureComponent {
    render() {
        return <React.Fragment>
            <h1>Hello world</h1>
        </React.Fragment>;
    }
}

export default PureComponentName;
`,
    {
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});


it('rawfunction snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component } from 'react';

function SubComponent() {
  return <div>Sub</div>;
}

const componentName = () => {
  return <div>
    <SubCoponent />
  </div>;
};

export default componentName;
`,
    {
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});


it('arrow-fragment snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component, Fragment } from 'react';

const componentName = () => {
  return <Fragment>
    <h1>Hello world</h1>
  </Fragment>;
};

export default componentName;
`,
    {
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});


it('arrow-react-fragment snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component } from 'react';

const componentName = () => {
  return <React.Fragment>
    <h1>Hello world</h1>
  </React.Fragment>;
};

export default componentName;
`,
    {
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});


it('nonJSX snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component } from 'react';

class TestClass extends Component {
  test() {
    return true;
  }
}

export default TestClass;
`,
    {
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});


it('pureComponent-fragment snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Fragment } from 'react';

class PureComponentName extends React.PureComponent {
    render() {
        return <Fragment>
            <h1>Hello world</h1>
        </Fragment>;
    }
}

export default PureComponentName;
`,
    {
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});


it('arrow-anonymous-fragment snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component, Fragment } from 'react';

const componentName = () => {
  return (() => <Fragment>
    <h1>Hello world</h1>
  </Fragment>)();
};

export default componentName;
`,
    {
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});


it('pure snapshot matches', () => {
  const { code } = babel.transform(
`import React from 'react';

class PureComponentName extends React.PureComponent {
    render() {
        return <div>
            <h1>Hello world</h1>
        </div>;
    }
}

export default PureComponentName;
`,
    {
      presets: ["@babel/preset-react"],
      plugins: [plugin]
    },
  );
  expect(code).toMatchSnapshot();
});


it('component-fragment-native snapshot matches', () => {
  const { code } = babel.transform(
`import React, { Component, Fragment } from 'react';

class componentName extends Component {
  render() {
    return <Fragment>A</Fragment>;
  }
}

export default componentName;
`,
    {
      presets: ["@babel/preset-react"],
      plugins: [
        [plugin, { native: true }]
      ]
    },
  );
  expect(code).toMatchSnapshot();
});


it('pure-native snapshot matches', () => {
  const { code } = babel.transform(
`import React from 'react';

class PureComponentName extends React.PureComponent {
    render() {
        return <div>
            <h1>Hello world</h1>
        </div>;
    }
}

export default PureComponentName;
`,
    {
      filename: "./filename-test.js",
      presets: ["@babel/preset-react"],
      plugins: [
        [plugin, { native: true }]
      ]
    },
  );
  expect(code).toMatchSnapshot();
});
