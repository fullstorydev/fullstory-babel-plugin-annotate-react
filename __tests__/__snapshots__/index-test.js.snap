// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrow snapshot matches 1`] = `
"import React, { Component } from 'react';

const componentName = () => {
  return /*#__PURE__*/React.createElement(\\"div\\", {
    \\"data-element\\": \\"div\\",
    \\"data-component\\": \\"componentName\\"
  }, /*#__PURE__*/React.createElement(\\"h1\\", {
    \\"data-element\\": \\"h1\\"
  }, \\"Hello world\\"));
};

export default componentName;"
`;

exports[`arrow-anonymous-fragment snapshot matches 1`] = `
"import React, { Component, Fragment } from 'react';

const componentName = () => {
  return (() => /*#__PURE__*/React.createElement(Fragment, null, /*#__PURE__*/React.createElement(\\"h1\\", null, \\"Hello world\\")))();
};

export default componentName;"
`;

exports[`arrow-anonymous-react-fragment snapshot matches 1`] = `
"import React, { Component } from 'react';

const componentName = () => {
  return (() => /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(\\"h1\\", null, \\"Hello world\\")))();
};

export default componentName;"
`;

exports[`arrow-fragment snapshot matches 1`] = `
"import React, { Component, Fragment } from 'react';

const componentName = () => {
  return /*#__PURE__*/React.createElement(Fragment, {
    \\"data-element\\": \\"Fragment\\",
    \\"data-component\\": \\"componentName\\"
  }, /*#__PURE__*/React.createElement(\\"h1\\", {
    \\"data-element\\": \\"h1\\"
  }, \\"Hello world\\"));
};

export default componentName;"
`;

exports[`arrow-noreturn snapshot matches 1`] = `
"import React, { Component } from 'react';

const componentName = () => /*#__PURE__*/React.createElement(\\"div\\", {
  \\"data-element\\": \\"div\\",
  \\"data-component\\": \\"componentName\\"
}, /*#__PURE__*/React.createElement(\\"h1\\", {
  \\"data-element\\": \\"h1\\"
}, \\"Hello world\\"));

export default componentName;"
`;

exports[`arrow-noreturn-fragment snapshot matches 1`] = `
"import React, { Component, Fragment } from 'react';

const componentName = () => /*#__PURE__*/React.createElement(Fragment, {
  \\"data-element\\": \\"Fragment\\",
  \\"data-component\\": \\"componentName\\",
  \\"data-source-file\\": \\"filename-test.js\\"
}, /*#__PURE__*/React.createElement(\\"h1\\", {
  \\"data-element\\": \\"h1\\"
}, \\"Hello world\\"));

export default componentName;"
`;

exports[`arrow-noreturn-react-fragment snapshot matches 1`] = `
"import React, { Component } from 'react';

const componentName = () => /*#__PURE__*/React.createElement(React.Fragment, {
  \\"data-element\\": \\"unknown\\",
  \\"data-component\\": \\"componentName\\"
}, /*#__PURE__*/React.createElement(\\"h1\\", {
  \\"data-element\\": \\"h1\\"
}, \\"Hello world\\"));

export default componentName;"
`;

exports[`arrow-react-fragment snapshot matches 1`] = `
"import React, { Component } from 'react';

const componentName = () => {
  return /*#__PURE__*/React.createElement(React.Fragment, {
    \\"data-element\\": \\"unknown\\",
    \\"data-component\\": \\"componentName\\"
  }, /*#__PURE__*/React.createElement(\\"h1\\", {
    \\"data-element\\": \\"h1\\"
  }, \\"Hello world\\"));
};

export default componentName;"
`;

exports[`component snapshot matches 1`] = `
"import React, { Component } from 'react';

class componentName extends Component {
  render() {
    return /*#__PURE__*/React.createElement(\\"div\\", {
      \\"data-element\\": \\"div\\",
      \\"data-component\\": \\"componentName\\"
    }, /*#__PURE__*/React.createElement(\\"h1\\", {
      \\"data-element\\": \\"h1\\"
    }, \\"Hello world\\"));
  }

}

export default componentName;"
`;

exports[`component-fragment snapshot matches 1`] = `
"import React, { Component, Fragment } from 'react';

class componentName extends Component {
  render() {
    return /*#__PURE__*/React.createElement(Fragment, {
      \\"data-element\\": \\"Fragment\\",
      \\"data-component\\": \\"componentName\\",
      \\"data-source-file\\": \\"filename-test.js\\"
    }, \\"A\\");
  }

}

export default componentName;"
`;

exports[`component-fragment-native snapshot matches 1`] = `
"import React, { Component, Fragment } from 'react';

class componentName extends Component {
  render() {
    return /*#__PURE__*/React.createElement(Fragment, {
      dataElement: \\"Fragment\\",
      dataComponent: \\"componentName\\"
    }, \\"A\\");
  }

}

export default componentName;"
`;

exports[`component-react-fragment snapshot matches 1`] = `
"import React, { Component } from 'react';

class componentName extends Component {
  render() {
    return /*#__PURE__*/React.createElement(React.Fragment, {
      \\"data-element\\": \\"unknown\\",
      \\"data-component\\": \\"componentName\\",
      \\"data-source-file\\": \\"filename-test.js\\"
    }, \\"A\\");
  }

}

export default componentName;"
`;

exports[`nonJSX snapshot matches 1`] = `
"import React, { Component } from 'react';

class TestClass extends Component {
  test() {
    return true;
  }

}

export default TestClass;"
`;

exports[`option-attribute snapshot matches 1`] = `
"import React, { Component } from 'react';

const componentName = () => {
  return /*#__PURE__*/React.createElement(\\"div\\", {
    \\"data-element\\": \\"div\\",
    \\"data-component\\": \\"componentName\\"
  }, /*#__PURE__*/React.createElement(\\"h1\\", {
    \\"data-element\\": \\"h1\\"
  }, \\"Hello world\\"));
};

export default componentName;"
`;

exports[`option-format snapshot matches 1`] = `
"import React, { Component } from 'react';

const componentName = () => {
  return /*#__PURE__*/React.createElement(\\"div\\", {
    \\"data-element\\": \\"div\\",
    \\"data-component\\": \\"componentName\\"
  }, /*#__PURE__*/React.createElement(\\"h1\\", {
    \\"data-element\\": \\"h1\\"
  }, \\"Hello world\\"));
};

export default componentName;"
`;

exports[`pure snapshot matches 1`] = `
"import React from 'react';

class PureComponentName extends React.PureComponent {
  render() {
    return /*#__PURE__*/React.createElement(\\"div\\", {
      \\"data-element\\": \\"div\\",
      \\"data-component\\": \\"PureComponentName\\"
    }, /*#__PURE__*/React.createElement(\\"h1\\", {
      \\"data-element\\": \\"h1\\"
    }, \\"Hello world\\"));
  }

}

export default PureComponentName;"
`;

exports[`pure-native snapshot matches 1`] = `
"import React from 'react';

class PureComponentName extends React.PureComponent {
  render() {
    return /*#__PURE__*/React.createElement(\\"div\\", {
      dataElement: \\"div\\",
      dataComponent: \\"PureComponentName\\",
      dataSourceFile: \\"filename-test.js\\"
    }, /*#__PURE__*/React.createElement(\\"h1\\", {
      dataElement: \\"h1\\"
    }, \\"Hello world\\"));
  }

}

export default PureComponentName;"
`;

exports[`pureComponent-fragment snapshot matches 1`] = `
"import React, { Fragment } from 'react';

class PureComponentName extends React.PureComponent {
  render() {
    return /*#__PURE__*/React.createElement(Fragment, {
      \\"data-element\\": \\"Fragment\\",
      \\"data-component\\": \\"PureComponentName\\"
    }, /*#__PURE__*/React.createElement(\\"h1\\", {
      \\"data-element\\": \\"h1\\"
    }, \\"Hello world\\"));
  }

}

export default PureComponentName;"
`;

exports[`pureComponent-react-fragment snapshot matches 1`] = `
"import React from 'react';

class PureComponentName extends React.PureComponent {
  render() {
    return /*#__PURE__*/React.createElement(React.Fragment, {
      \\"data-element\\": \\"unknown\\",
      \\"data-component\\": \\"PureComponentName\\"
    }, /*#__PURE__*/React.createElement(\\"h1\\", {
      \\"data-element\\": \\"h1\\"
    }, \\"Hello world\\"));
  }

}

export default PureComponentName;"
`;

exports[`rawfunction snapshot matches 1`] = `
"import React, { Component } from 'react';

function SubComponent() {
  return /*#__PURE__*/React.createElement(\\"div\\", {
    \\"data-element\\": \\"div\\",
    \\"data-component\\": \\"SubComponent\\"
  }, \\"Sub\\");
}

const componentName = () => {
  return /*#__PURE__*/React.createElement(\\"div\\", {
    \\"data-element\\": \\"div\\",
    \\"data-component\\": \\"componentName\\"
  }, /*#__PURE__*/React.createElement(SubCoponent, {
    \\"data-element\\": \\"SubCoponent\\"
  }));
};

export default componentName;"
`;

exports[`rawfunction-fragment snapshot matches 1`] = `
"import React, { Component, Fragment } from 'react';

function SubComponent() {
  return /*#__PURE__*/React.createElement(Fragment, {
    \\"data-element\\": \\"Fragment\\",
    \\"data-component\\": \\"SubComponent\\"
  }, \\"Sub\\");
}

const componentName = () => {
  return /*#__PURE__*/React.createElement(Fragment, {
    \\"data-element\\": \\"Fragment\\",
    \\"data-component\\": \\"componentName\\"
  }, /*#__PURE__*/React.createElement(SubCoponent, {
    \\"data-element\\": \\"SubCoponent\\"
  }));
};

export default componentName;"
`;

exports[`rawfunction-react-fragment snapshot matches 1`] = `
"import React, { Component } from 'react';

function SubComponent() {
  return /*#__PURE__*/React.createElement(React.Fragment, {
    \\"data-element\\": \\"unknown\\",
    \\"data-component\\": \\"SubComponent\\"
  }, \\"Sub\\");
}

const componentName = () => {
  return /*#__PURE__*/React.createElement(React.Fragment, {
    \\"data-element\\": \\"unknown\\",
    \\"data-component\\": \\"componentName\\"
  }, /*#__PURE__*/React.createElement(SubCoponent, {
    \\"data-element\\": \\"SubCoponent\\"
  }));
};

export default componentName;"
`;

exports[`tags snapshot matches 1`] = `
"import React, { Component } from 'react';
import { StyleSheet, Text, TextInput, View, Image, UIManager } from 'react-native';
UIManager.getViewManagerConfig('RCTView').NativeProps.fsClass = \\"String\\";

class Bananas extends Component {
  render() {
    let pic = {
      uri: 'https://upload.wikimedia.org/wikipedia/commons/d/de/Bananavarieties.jpg'
    };
    return /*#__PURE__*/React.createElement(Image, {
      source: pic,
      style: {
        width: 193,
        height: 110,
        marginTop: 10
      },
      fsClass: \\"test-class\\",
      dataElement: \\"Image\\",
      dataComponent: \\"Bananas\\",
      dataSourceFile: \\"filename-test.js\\"
    });
  }

}

class PizzaTranslator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      text: ''
    };
  }

  render() {
    return /*#__PURE__*/React.createElement(View, {
      style: {
        padding: 10
      },
      dataElement: \\"View\\",
      dataComponent: \\"PizzaTranslator\\",
      dataSourceFile: \\"filename-test.js\\"
    }, /*#__PURE__*/React.createElement(TextInput, {
      style: {
        backgroundColor: '#000',
        color: '#eee',
        padding: 8
      },
      placeholder: \\"Type here to translate!\\" // not supported on iOS
      ,
      onChangeText: text => this.setState({
        text
      }),
      value: this.state.text,
      dataElement: \\"TextInput\\"
    }), /*#__PURE__*/React.createElement(Text, {
      style: {
        padding: 10,
        fontSize: 42
      },
      dataElement: \\"Text\\"
    }, this.state.text.split(' ').map(word => word && '🍕').join(' ')));
  }

}

export default function App() {
  return /*#__PURE__*/React.createElement(View, {
    style: styles.container,
    dataElement: \\"View\\",
    dataComponent: \\"App\\",
    dataSourceFile: \\"filename-test.js\\"
  }, /*#__PURE__*/React.createElement(Text, {
    style: {
      color: '#eee'
    },
    dataElement: \\"Text\\"
  }, \\"FullStory ReactNative testing app\\"), /*#__PURE__*/React.createElement(Bananas, {
    dataElement: \\"Bananas\\"
  }), /*#__PURE__*/React.createElement(PizzaTranslator, {
    dataElement: \\"PizzaTranslator\\"
  }));
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'stretch',
    backgroundColor: '#222',
    alignItems: 'center',
    justifyContent: 'center'
  }
});"
`;
